paasta_tools/autoscaling/autoscaling_cluster_lib.py

```diff
@@ -178,20 +219,26 @@ def is_aws_launching_sfr_instances(sfr):
     return target_capacity > fulfilled_capacity
 
 
-def cleanup_cancelled_sfr_config(spotfleet_request_id, config_folder, dry_run=False):
-    sfr_file_name = "{0}.json".format(spotfleet_request_id)
-    sfr_configs_to_delete = [os.path.join(walk[0], sfr_file_name)
-                             for walk in os.walk(config_folder) if sfr_file_name in walk[2]]
-    if not sfr_configs_to_delete:
-        log.info("SFR config for {0} not found".format(spotfleet_request_id))
+def is_aws_launching_asg_instances(asg):
+    fulfilled_capacity = len(asg['Instances'])
+    target_capacity = asg['DesiredCapacity']
+    return target_capacity > fulfilled_capacity
+
+
+def cleanup_cancelled_config(resource_id, config_folder, dry_run=False):
+    file_name = "{0}.json".format(resource_id)
+    configs_to_delete = [os.path.join(walk[0], file_name)
+                         for walk in os.walk(config_folder) if file_name in walk[2]]
+    if not configs_to_delete:
+        log.info("Resource config for {0} not found".format(resource_id))
         return
     if not dry_run:
-        os.remove(sfr_configs_to_delete[0])
-    log.info("Deleted SFR config {0}".format(sfr_configs_to_delete[0]))
+        os.remove(configs_to_delete[0])
+    log.info("Deleted Resource config {0}".format(configs_to_delete[0]))
 
 
-def get_sfr_slaves(resource, mesos_state):
-    instance_ips = get_sfr_instance_ips(resource['sfr'], region=resource['region'])
+def get_aws_slaves(resource, mesos_state):
```

Comment:
this seems kind of muddled: this isn't really getting aws slaves, but just matching slaves by the ips in a resource, right? so this won't return the complete set of aws slaves in the mesos_state; just the ones in this resource. Make this a bit more generic:

mesos_tools.py
```
def filter_slaves_by_ip(acceptable_ips, slaves):
   return filter(lambda slave: slave_pid_to_ip(slave['pid') in acceptable_ips)
```

autoscaling_lib.py
```
def get_slaves_in_resource(resource, slaves):
    instance_ips = get_instance_ips(resource['ActiveInstances'], region=resource['region'])
    return mesos_tools.filter_slaves_by_ip(instance_ips, slaves)
```

I also don't understand why this is creating a dict of id: slave - we already have the ids in the slave object, why do we need to make a more complicated data structure?